# For move_base
base_global_planner: "SBPLLatticePlanner"
#base_global_planner: "navfn/NavfnROS"
#base_global_planner: "global_planner/GlobalPlanner"
#base_global_planner: "limited_orientation_global_planner/LimitedOrientationGlobalPlanner"

#recovery_behaviors: [{name: conservative_reset, type: clear_costmap_recovery/ClearCostmapRecovery}, {name: aggressive_reset, type: clear_costmap_recovery/ClearCostmapRecovery}] 
recovery_behaviors: [] 
recovery_behavior_enabled: false # true

# For move_base_flex
planners:
  - name: 'SBPLLatticePlanner'
    type: 'SBPLLatticePlanner'
  - name: 'WaypointGlobalPlanner'
    type: 'waypoint_global_planner/WaypointGlobalPlanner'
  - name: 'NoObstaclesWaypointGlobalPlanner'
    type: 'waypoint_global_planner/WaypointGlobalPlanner'

#recovery_behaviors:
#  - name: 'clear_costmap'
#    type: 'clear_costmap_recovery/ClearCostmapRecovery'


# Oscillation
oscillation_timeout: 50.0
oscillation_distance: 0.5

controller_frequency: 5
controller_patience: 1
planner_frequency: 1
planner_patience: 0.01
max_planning_retries: 0

NavfnROS:
  allow_unknown: true  # TODO: investigate this
  default_tolerance: 0.0
  # do not restrict planner
  planner_window_x: 0.0
  planner_window_y: 0.0
  # debug
  visualize_potential: false

GlobalPlanner:
  allow_unknown: true
  visualize_potential: true
  use_dijkstra: false
  use_grid_path: true

LimitedOrientationGlobalPlanner:
  allow_unknown: true
  visualize_potential: true
  use_dijkstra: false
  use_grid_path: true

SBPLLatticePlanner:
  allocated_time: 15.0
  lethal_obstacle: 150.0

WaypointGlobalPlanner:
  lethal_cost: 150.0
  waypoints_per_meter: 20
  waypoints_per_radiant: 10
  max_collision_check_radius: 10

# Configuration of the WaypointGlobalPlanner to ignore obstacles
# it also uses a lower density of waypoints
NoObstaclesWaypointGlobalPlanner:
  lethal_cost: 256
  waypoints_per_meter: 1
  waypoints_per_radiant: 1
  max_collision_check_radius: 0
